invoices/invoiceregistry.go

```diff
@@ -998,6 +1004,49 @@ func (i *InvoiceRegistry) notifyExitHopHtlcLocked(
 	)
 
 	callback := func(inv *Invoice) (*InvoiceUpdateDesc, error) {
+		// Provide the invoice to the settlement interceptor to allow
+		// the interceptor's client an opportunity to settle the
+		// invoice.
+		var response fn.Option[InterceptSession]
+		if i.cfg.SettlementInterceptor != nil {
+			response = i.cfg.SettlementInterceptor.HandleInvoice(
+				*inv,
+			)
+		}
+
+		// If the interceptor service has provided a response, we'll
+		// wait for the client to respond with a resolution.
+		if response.IsSome() {
+			session := response.UnsafeFromSome()
+
+			select {
+			case r := <-session.ResolutionChan:
+				// If the interceptor has provided a resolution,
+				// we'll use that as the final resolution.
+				resolution = r.HtlcResolution
+
+				var updateDesc *InvoiceUpdateDesc
+				r.InvoiceUpdateDesc.WhenSome(
+					func(desc InvoiceUpdateDesc) {
+						updateDesc = &desc
+					},
+				)
+
+				// Only send an update if the invoice state was
+				// changed.
+				updateSubscribers = updateDesc != nil &&
+					updateDesc.State != nil
+
+				return updateDesc, nil
```

Comment:
So we still want to call into `updateInvoice` below as normal. That has other logic to do things like check the invoice expiry, etc. 

For this layer, we just want a simple predicate if the invoice should be accepted or not based on the invoice, and also the extended HTLCs for the invoice.

Reply:
I've moved the interceptor derived resolution injection to occur after the `updateInvoice` call. I think that's what you mean. Please let me know if I've miss understood.